### PR TITLE
feat: add lint and formatting commit hooks

### PR DESCRIPTION
**Pull request by lisunshiny**
Addresses #30

This CL adds 3 new GitHub workflows (but only 2 are enabled)
* A hook to run linting (https://github.com/terraform-linters/tflint) which checks for unused variables, etc.
* A hook to run terraform fmt (https://github.com/dflook/terraform-github-actions/tree/main/terraform-fmt-check) and fail if files are not properly formatted
* A hook that I commented out to show the output of `terraform plan`. This seems super useful but is not related to formatting and is probably more CI/CD related vs. just enforcing formatting. Happy to just delete the file if we don't think we'll use this at all. 

Note I will squash merge given that my commit messages are... creative.

### CODE REVIEW

**Comment by lisunshiny** [comment]
Could someone with more access than me (@phattya ?) "approve and run" the workflows in this PR if that exists? 
 https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

**Review by phattya — APPROVED** [review]
@lisunshiny I actually don't see step 5 as available to me either, in that link you sent. Maybe another review has to happen first for it to be available?  This all looks good to me, but definitely need another review from others...and maybe @chriscasto in particular, since I'm still learning.  I'm approving now anyway, since we can always revert/change and this will unblock you.

image

**Comment by glgh** [comment]
nice! think this will be a very helpful reference for some of the CI/CD stuff

**Pull request merged**
